### ISSUE TITLE
Patch doesn't work with Nvidia cards on hackingtosh.   macOS Sierra

### ISSUE THREAD

**Issue by svickory** [Floris497/mac-pixel-clock-patch-V2#88]
just did a fresh install and loaded. Nvidia web drivers.  reboot. everything working install the coredisplay patch. reboot.  install SwitchresX.  I see. 3840*2160 60hz. I choose it screen goes blanks.  I am running dual monitor so I have a monitor in dvi Port which is still working. choose 3840x2160 30Hz.  monitor comes back on..  Everything works find in Elcapitan

**Comment by Floris497** [comment]
Use the IOKit patch from the test branch for 10.12, unpatch CoreDisplay. let me know if that works :)

**Comment by svickory** [comment]
Nope Same Issue.

**Comment by Floris497** [comment]
Used the one from here? https://github.com/Floris497/mac-pixel-clock-patch-V2/tree/Siera-nvidia-IOKit-test

And if you patch both IOKit and CoreDisplay?
Maybe fully uninstall everything from Switchresx patch and reinstall..

**Comment by svickory** [comment]
Yes..  Same Results  when I did IOKIT  got unknown Version also

**Comment by Floris497** [comment]
Ah.. should not be unknown.. Could you send me the md5 output from IOKit patch? `IOKit-patch.command md5`

**Comment by d3funkt** [comment]
Hello Floris497,

I decided to take part in the beta test with: 

- a fresh install of Sierra 10.12 16A323
- a GTX 970, Web Driver 367.15.10.05f01
- HDMI > HDMI connected to Sony XBR43X800D 

With this setup I can only achieve **4k30p in Sierra**
**4k60p in Win 10**

**Here are the IOKit results:**

> Last login: Tue Oct 11 03:09:41 on console
> Name:~ account$ /Users/account/Downloads/mac-pixel-clock-patch-V2-Siera-nvidia-IOKit-test/IOKit-patcher.command 
> SIP looks to be disabled, all good!
> 
> Unknown version of IOKit found..
> ---- BEGINNING MD5 HASH SUMS ---- version: 10.12 build:16A323
> 
>   otool IOKit: 1663d5e916f7cec456ee87117f39b9df
>  otool IOKit.bak: NO FILE (this is okay)
>            IOKit: 467e112c5c925cfe92a0a0f5e8242c74
>        IOKit.bak: NO FILE (this is okay)
> 
> ---- ENDING MD5 HASH SUMS -------
> 

**I decided to force patch v7:**

> Name:~ account$ /Users/account/Downloads/mac-pixel-clock-patch-V2-Siera-nvidia-IOKit-test/IOKit-patcher.command patch v7
> SIP looks to be disabled, all good!
> Patching IOKit with patch version 7
> Password:
> Re-singing /System/Library/Frameworks/IOKit.framework/Versions/A/IOKit
> /System/Library/Frameworks/IOKit.framework/Versions/A/IOKit: replacing existing signature
> The code of IOKit changed, the patch was probbably succesfull
> Name:~ account$ 

**No luck.** 

Patching CoreDisplay has the effect of enabling 50hz and 60hz options, but when set to Scaled 3840 x 2160 50hz or 60hz, the screen goes "black" aka no signal.
Unpatching CoreDisplay makes the 50hz and 60hz options disappear.

**Comment by svickory** [comment]
I have the same Issue D3funkt

**Comment by baronomasia** [comment]
+1 to this issue. I'm in a similar boat - upgraded to Sierra from El Capitan with a working IOKit patch for 4k@60Hz, but once CoreDisplay was patched after the Sierra update, on boot, the screen goes black just before the login screen should appear. I reverted the patch and everything goes back to 30 Hz, but no luck on 60 Hz :(. I haven't attempted the experimental IOKit patch, nor have I installed SwitchResX (which wasn't necessary in El Capitan), but I suspect if I attempt I'll have the same luck as everyone else in this thread. In case it is helpful, a couple of specs are below:

i5-3570k
NVIDIA 980 Ti

I'd be happy to test any experimental changes. I have no problems booting into recovery and correcting things if need be. Thanks for your work!

**Comment by vladshved** [comment]
+1 Same here, using GTX 960. El Capitan is buttery smooth at 60, and Sierra goes to black screen. Removing patch and its cruising at 30hz just fine. Thank you!

**Comment by Edeoinc** [comment]
Same issue here with  GTX 970. 
4k 60hz won't show any image on Sierra, same hardware on Windows works fine.
Will be glad to help you somehow to resolve that issue. Thanks.

**Comment by Floris497** [comment]
What cable do you use and what type of display?

**Comment by Edeoinc** [comment]
Phillips HDMI 2.0v cable.
Samsung ku6300 display. It has 1 hdmi port capable of 4k 60hz 4:4:4.

**Comment by Floris497** [comment]
Windows tents to do 4:2:0 to make 60Hz possible on HDMI 1.4 What does SwitchResX say?

**Comment by Edeoinc** [comment]
Yeah, i know about that windows behavior, so i forced 4:4:4 in driver.
![screen shot 2016-10-29 at 6 36 56 pm](https://cloud.githubusercontent.com/assets/11458376/19830837/c7bf2ada-9e06-11e6-8850-03bcc42a304e.png)
![screen shot 2016-10-29 at 6 36 47 pm](https://cloud.githubusercontent.com/assets/11458376/19830836/c7be80e4-9e06-11e6-95dd-8a01146317d8.png)

**Comment by Floris497** [comment]
Oh.. wait.. you still need the IOKit patch maybe.. could you check if the one in the Nvidia test branch works for you?

**Comment by Floris497** [comment]
Just updated it, I think it should work now without forcing it. looks like the patch worked fine for me. no word on HDMI2.0 tho. I don't have a working HDMI2.0 Video card..

**Comment by Edeoinc** [comment]
I've tried new IOKit Patcher on top of CoreDisplay Patcher, still no 4k 60hz, and 60hz option disappeared.
Unpatch both, and reapply patches didn't work too.
I guess ill try one more time later on clean system.
Big thanks for your time!

**Comment by Floris497** [comment]
Okay, let me know if you find something. maybe there is something in the issues here that can help you. It's a bit unorganised here and I cannot remember all info from here.. No problem :)
